src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -5105,16 +5106,42 @@ protected boolean indexAtEitherBounds(RubyString string, int index) {
 
     }
 
+    @ImportStatic(RopeGuards.class)
     @Primitive(name = "string_to_inum", lowerFixnum = 1)
-    public abstract static class StringToInumPrimitiveNode extends PrimitiveArrayArgumentsNode {
+    @NodeChild(value = "string", type = RubyNode.class)
+    @NodeChild(value = "fixBase", type = RubyNode.class)
+    @NodeChild(value = "strict", type = RubyNode.class)
+    @NodeChild(value = "raiseOnError", type = RubyNode.class)
+    public abstract static class StringToInumPrimitiveNode extends PrimitiveNode {
+
+        @Child FixnumOrBignumNode fixnumOrBignumNode;
+        @Child RopeNodes.BytesNode bytesNode;
+
+        @CreateCast("string")
+        protected RubyNode coerceStringToRope(RubyNode string) {
+            return ToRopeNodeGen.create(string);
+        }
+
+        @Specialization(guards = {
+                "isLazyIntRope(rope)",
+                "doesNotRequireBaseConversion(fixBase)"
+        })
+        protected Object stringToInumIntRope(Rope rope, int fixBase, boolean strict, boolean raiseOnError) {
+            return ((LazyIntRope) rope).getValue();
+        }
 
         @Specialization
-        protected Object stringToInum(Object string, int fixBase, boolean strict, boolean raiseOnError,
-                @Cached("new()") FixnumOrBignumNode fixnumOrBignumNode,
-                @Cached RopeNodes.BytesNode bytesNode,
-                @Cached BranchProfile exceptionProfile,
-                @CachedLibrary(limit = "2") RubyStringLibrary libString) {
-            final Rope rope = libString.getRope(string);
+        protected Object stringToInum(Rope rope, int fixBase, boolean strict, boolean raiseOnError,
+                @Cached BranchProfile exceptionProfile) {
+            if (bytesNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                bytesNode = insert(RopeNodes.BytesNode.create());
+            }
+
+            if (fixnumOrBignumNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                fixnumOrBignumNode = insert(new FixnumOrBignumNode());
+            }
```

Comment:
`@Cached` is preferred to the lazy pattern if only used in one specialization, could you move those 2 nodes back to `@Cached`?

Reply:
Will do. Sorry about that. I had three specializations at one point, but backed one out for now.